src/factuality_eval/dataset_generation.py

```diff
@@ -230,3 +235,23 @@ def generate_hash(context: list[str], question: str, answer: str) -> str:
         A unique hash string for the QA pair.
     """
     return hashlib.md5((context[0] + question + answer).encode("utf-8")).hexdigest()
+
+
+def get_hallucinated_labels(hallucinated_dict: dict) -> list[dict]:
+    """Get the hallucinated labels from the generation result.
+
+    Args:
+        hallucinated_dict:
+            The dictionary from the hallucination generator.
+
+    Returns:
+        A list of dictionaries with start, end, and label for each hallucinated part.
+    """
+    hallucinated_labels = []
+    for part in hallucinated_dict["hallucinated_parts"]:
+        start = hallucinated_dict["hallucinated_answer"].find(part)
```

Comment:
Maybe include a sanity check here in case `part` appears multiple times in the hallucinated answer:

```suggestion
        if hallucinated_dict["hallucinated_answer"].count(part) > 1:
            raise ValueError(
                f"The part {part!r} appears multiple times in the hallucinated answer "
                f"{hallucinated_dict['hallucinated_answer']!r}, so could not correctly "
                "mark the spans."
            )
        start = hallucinated_dict["hallucinated_answer"].find(part)
```